### PR TITLE
papr2kube: Experiment to convert papr to Kubernetes jobs

### PR DESCRIPTION
**Pull request by cgwalters**
After setup:
```
oc cluster up
...
oc adm policy add-scc-to-group anyuid system:authenticated
```

```
./papr/papr2kube.py --limit 1 ostreedev/ostree 886a5d79280b92d336a4eecc9da287fa8b697542 ~/src/github/ostreedev/ostree/.papr.yml  > ostree-papr-kube.yml
oc create -f ostree-papr-kube.yml
```

Then I can e.g. watch a build with:
`oc logs -f jobs/papr-ostreedev-ostree-886a5d7928-0`

### CODE REVIEW

**Comment by cgwalters** [comment]
More info on [Kubernetes jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/).

**Comment by cgwalters** [comment]
To support virt, we could build custom containers with the qcow2 images.  Then we don't need OpenStack anymore.

**Comment by ashcrow** [comment]
Very cool @cgwalters!!! 👍 

The "failures" in CI are related to ``flake8`` checks:
```
papr/papr2kube.py:8:1: F401 'time' imported but unused
papr/papr2kube.py:10:1: F401 'traceback' imported but unused
papr/papr2kube.py:12:1: F401 'subprocess' imported but unused
papr/papr2kube.py:15:1: F401 'papr.PKG_DIR' imported but unused
papr/papr2kube.py:20:1: E302 expected 2 blank lines, found 1
papr/papr2kube.py:22:80: E501 line too long (80 > 79 characters)
papr/papr2kube.py:34:14: E201 whitespace after '{'
papr/papr2kube.py:36:13: E124 closing bracket does not match visual indentation
papr/papr2kube.py:43:10: E201 whitespace after '{'
papr/papr2kube.py:46:16: E201 whitespace after '{'
papr/papr2kube.py:48:15: E124 closing bracket does not match visual indentation
papr/papr2kube.py:54:80: E501 line too long (87 > 79 characters)
papr/papr2kube.py:56:80: E501 line too long (110 > 79 characters)
papr/papr2kube.py:57:9: E124 closing bracket does not match visual indentation
papr/papr2kube.py:73:5: E124 closing bracket does not match visual indentation
papr/papr2kube.py:76:1: E302 expected 2 blank lines, found 1
papr/papr2kube.py:79:80: E501 line too long (88 > 79 characters)
papr/papr2kube.py:80:80: E501 line too long (88 > 79 characters)
papr/papr2kube.py:83:80: E501 line too long (94 > 79 characters)
papr/papr2kube.py:86:9: E231 missing whitespace after ','
papr/papr2kube.py:94:80: E501 line too long (88 > 79 characters)
papr/papr2kube.py:99:10: E231 missing whitespace after ','
papr/papr2kube.py:107:1: E305 expected 2 blank lines after class or function definition, found 1
papr/utils/os_provision.py:163:5: E722 do not use bare except'
```

File: papr/papr2kube.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          'volumeMounts': [
+              { 'name': 'builddir',
+                'mountPath': '/srv',
+              }
+          ],
+          'securityContext': {'runAsUser': 0},
+          'workingDir': '/srv/',
+          'command': ['/bin/sh', '-c',
+                      '''set -xeuo pipefail; yum -y install git
+                      git clone --depth=100 https://github.com/{gh_org}/{gh_repo} build
```
Comment (cgwalters) [inline papr/papr2kube.py:54]: Obviously this is a hack; we should at least optionally support a git mirroring setup (itself exposed as a kube service).

**Comment by cgwalters** [comment]
And of course going this route conflicts with https://github.com/projectatomic/papr/pull/58 in the sense that a lot of code would be fundamentally different.

**Comment by jlebon** [comment]
Ouhh, very nice! Will try this out later!

File: papr/papr2kube.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    ]
+    containers = [{
+        'name': name,
+        'image': suite['container']['image'],
+        'volumeMounts': [
+            { 'name': 'builddir',
+              'mountPath': '/srv',
+            }
+        ],
+        'securityContext': {'runAsUser': 0},
```
Comment (jlebon) [inline papr/papr2kube.py:38]: This is something we'll have to solve/workaround if we want to be able to run on most (all?) managed OCP instances though, right? I suppose we can do like Travis and provide the option between unprivileged and privileged tests, with the latter just resulting in spinning up a light VM?

Comment (cgwalters) [reply papr/papr2kube.py:38]: I'm sure that we can get privileges to do this in at least *some* OCP clusters, from internal OpenStack to CentOS CI.   Not to mention that at some point user namespaces are going to be plumbed through into Kube, and that will entirely change things here.

**Comment by cgwalters** [comment]
Let's discuss PAPR/ci/testing architecture at a meeting sometime soon?  One thread we can follow here is to try to move PAPR more fully into CentOS CI.  Then we'd naturally have discussions there.

**Comment by ashcrow** [comment]
@cgwalters / @jlebon / @dustymabe this would be a good design discussion for someone in the team to "own" and make sure discussions occur for this sprint if one hasn't been decided on yet.

**Comment by jlebon** [comment]
Sorry for not merging this! No reason to not have it in the main tree until we figure out the migration path to CCI.

**Pull request merged**